### ISSUE TITLE
Cross Compilation failed on RDK Toolchain with different build flags

### ISSUE THREAD

**Issue by kanjoe24** [rdkcentral/ut-control#37]
@kanjoe24  @Ulrond 
I have validated the PR#92 changes on 2 RDKB hals:
1. Cellular modem HAL : 
     Cross-compilation : Made changes mentioned in https://github.com/rdkcentral/ut-control/issues/34. triggered the build, attaching error logs observed: 
[cellularmodem_log_crossCompilation.txt](https://github.com/user-attachments/files/16439659/cellularmodem_log_crossCompilation.txt) ( Used  XLE tool chain for cross-compiling)

_Originally posted by @saniya-09 in https://github.com/rdkcentral/ut-control/issues/36#issuecomment-2260141467_

**Comment by Ulrond** [comment]
* need to check if this is a real isuse, or if this was just by swapping from linux to rdk toolchain a clean wasn't provided.
* If it turns out that the clean wasn't done, then we need a method for auto-performing the clean, or building the target tools into linux / arm directories so there's never an issue.

**Comment by kanjoe24** [comment]
Looks like a real issue as the curl configure gives clear list of libs that are required for each of the platform.
For MTA:
```
configure: Configured to build curl/libcurl:

  Host setup:       arm-rdk-linux-gnueabi
  Install prefix:   /home/ssebas862/saniya/L1OpenSourcing/31072024_mtahal_ut-core/rdkb-halif-mta/ut/ut-core/framework/ut-control/framework/curl/curl-8.8.0/build
  Compiler:         arm-rdk-linux-gnueabi-gcc  -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a15 --sysroot=/home/ssebas862/SDK_Toolchain/tool_TCXB7/sysroots/cortexa15hf-neon-rdk-linux-gnueabi
   CFLAGS:          -Os -pipe -g -feliminate-unused-debug-types -Werror-implicit-function-declaration -Wno-system-headers -pthread
   CPPFLAGS:        -isystem /home/ssebas862/saniya/L1OpenSourcing/31072024_mtahal_ut-core/rdkb-halif-mta/ut/ut-core/framework/ut-control/framework/openssl/openssl-OpenSSL_1_1_1w/build/include
   LDFLAGS:         -L/home/ssebas862/saniya/L1OpenSourcing/31072024_mtahal_ut-core/rdkb-halif-mta/ut/ut-core/framework/ut-control/framework/openssl/openssl-OpenSSL_1_1_1w/build/lib
   LIBS:            -lssl -lcrypto -lssl -lcrypto -lz

``` 

For CellularModem:
```
 Host setup:       arm-rdk-linux-gnueabi
  Install prefix:   /home/ssebas862/saniya/L1OpenSourcing/31072024_ut-core320/rdkb-halif-cellular-modem/ut/ut-core/framework/ut-control/framework/curl/curl-8.8.0/build
  Compiler:         arm-rdk-linux-gnueabi-gcc  -mthumb -mfpu=vfp  -mcpu=cortex-a9 -mfloat-abi=soft -mabi=aapcs-linux -mno-thumb-interwork -ffixed-r8 -fomit-frame-pointer --sysroot=/home/ssebas862/SDK_Toolchain/tool_XLE/tmp/sysroots/xle-wnc
   CFLAGS:          -Os -pipe -g -feliminate-unused-debug-types -DSUPPORT_IPV6 -DSKY -DBRCM_CMS_MSG -DFEATURE_RDKB_WAN_MANAGER -D_SKY_HUB_COMMON_PRODUCT_REQ_ -DFEATURE_RDKB_LED_MANAGER -D_WNXL11BWL_PRODUCT_REQ_ -DRDKB_EXTENDER -DRDKB_CELLULAR_SUPPORT -DCHIP_FAMILY_ID_HEX=0x6855 -DFEATURE_RDKB_NO_WPS_BUTTON -DGPIO_LED_MGR -DRBUS_BUILD_FLAG_ENABLE -DXBOOT_PLAT_PLATFORM_XLE -DXBOOT_PLAT_KEYSTORE_UPDATE -DXBOOT_PLAT_CERTTORE_UPDATE -DXBOOT_PLAT_BLDATA_UPDATE -DFEATURE_RDKB_THERMAL_MANAGER -Werror-implicit-function-declaration -Wno-system-headers -pthread
   CPPFLAGS:        -isystem /home/ssebas862/saniya/L1OpenSourcing/31072024_ut-core320/rdkb-halif-cellular-modem/ut/ut-core/framework/ut-control/framework/openssl/openssl-OpenSSL_1_1_1w/build/include
   LDFLAGS:         -L/home/ssebas862/saniya/L1OpenSourcing/31072024_ut-core320/rdkb-halif-cellular-modem/ut/ut-core/framework/ut-control/framework/openssl/openssl-OpenSSL_1_1_1w/build/lib
   LIBS:            -lidn2 -lpsl -lssl -lcrypto -lssl -lcrypto -lz

``` 
The libs specified in cellular modem are exactly the ones that causes the linking error and are currently missing in our makefile.

**Comment by kanjoe24** [comment]
Fixed as part of #36 .
